src/index.ts

```diff
@@ -82,22 +82,23 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
 
   const request = event.request
   const ASSET_NAMESPACE = options.ASSET_NAMESPACE
-  const ASSET_MANIFEST = options.ASSET_MANIFEST
+  const ASSET_MANIFEST = typeof (options.ASSET_MANIFEST) === 'string'
+    ? JSON.parse(options.ASSET_MANIFEST)
+    : options.ASSET_MANIFEST
 
-  const SUPPORTED_METHODS = ['GET', 'HEAD']
+  if (typeof ASSET_NAMESPACE === 'undefined') {
+    throw new InternalError(`there is no KV namespace bound to the script`)
+  }
 
+  const SUPPORTED_METHODS = ['GET', 'HEAD']
   if (!SUPPORTED_METHODS.includes(request.method)) {
     throw new MethodNotAllowedError(`${request.method} is not a valid request method`)
   }
 
-  if (typeof ASSET_NAMESPACE === 'undefined') {
-    throw new InternalError(`there is no KV namespace bound to the script`)
-  }
-
-  // determine the requestKey based on the actual file served for the incoming request
-  const requestKey = options.mapRequestToAsset(request)
+  const rawPathKey = new URL(request.url).pathname.replace(/^\/+/, '') // original request's pathname (e.g. foo/filename)
```

Comment:
is there any way we can do this with `.split` and other string manipulation patterns? regular expressions are hard to read and hard to debug

Reply:
unfortunately, even with the comment i'm not sure what this is supposed to do

Reply:
```suggestion
  const rawPathKey = new URL(request.url).pathname.replace(/^\/+/, '') // strip any preceding /'s 
```
Unfortunately split would only let us get the 1st slash and that logic would get funky too. How about this comment?

Reply:
yah sometimes a regex is the right way to do things; fortunately this isn't a particularly gnarly one and this version of the comment definitely helps.